src/Toolkit/Toolkit/UI/Controls/PopupViewer/UtilityAssociationGroupResultPopupView.Maui.cs

```diff
@@ -0,0 +1,173 @@
+            layout.Add(cv);
+
+            INameScope nameScope = new NameScope();
+            NameScope.SetNameScope(layout, nameScope);
+            nameScope.RegisterName("ClearSearch", clearButton);
+            nameScope.RegisterName("SearchText", searchText);
+            nameScope.RegisterName("ResultsList", cv);
+
+            return layout;
+        }
+
+        private static object BuildDefaultItemTemplate()
+        {
+            Grid layout = new Grid() { Margin = new Thickness(10, 0, 0, 0) };
+            TapGestureRecognizer itemTapGesture = new TapGestureRecognizer();
+            itemTapGesture.Tapped += ResultsListView_Tapped;
+            layout.GestureRecognizers.Add(itemTapGesture);
+
+            var view = new UtilityAssociationResultPopupView();
+            view.SetBinding(UtilityAssociationResultPopupView.AssociationResultProperty, static (UtilityNetworks.UtilityAssociationResult result) => result);
+            layout.Add(view);
+            return layout;
+        }
+
+        /// <inheritdoc />
+        protected override void OnApplyTemplate()
+        {
+            base.OnApplyTemplate();
+
+            if (_searchEntry is not null)
+            {
+                _searchEntry.TextChanged -= SearchText_TextChanged;
+            }
+
+            if (_clearSearchButton is not null)
+            {
+                _clearSearchButton.Clicked -= ClearSearchButton_Clicked;
+            }
+
+            if (GetTemplateChild("ResultsList") is CollectionView listView)
+            {
+                _resultsListView = listView;
+            }
+
+            if (GetTemplateChild("SearchText") is Entry entry)
+            {
+                _searchEntry = entry;
+                _searchEntry.TextChanged += SearchText_TextChanged;
+            }
+            if (GetTemplateChild("ClearSearch") is ImageButton button)
+            {
+                _clearSearchButton = button;
+                _clearSearchButton.Clicked += ClearSearchButton_Clicked;
+            }
+
+            UpdateView();
+        }
+
+        private void ClearSearchButton_Clicked(object? sender, EventArgs e)
+        {
+            if (_searchEntry is not null)
+            {
+                _searchEntry.Text = string.Empty;
+            }
+        }
+
+        private void UpdateView()
+        {
+            if (_resultsListView is not null)
+            {
+                _resultsListView.ItemsSource = GroupResult?.AssociationResults;
+            }
+        }
+
+        private void SearchText_TextChanged(object? sender, TextChangedEventArgs e)
+        {
+            var searchText = (sender as Entry)?.Text.Trim();
+            if (!string.IsNullOrWhiteSpace(searchText))
+            {
+                if (_resultsListView is not null)
+                {
+                    _resultsListView.ItemsSource = GroupResult?.AssociationResults?.Where(r => r.Title?.IndexOf(searchText, StringComparison.OrdinalIgnoreCase) >= 0);
```

Comment:
I think this filter could be made safe against NullReferenceException with additional checks and trimming of whitespaces in string similar to below.

```suggestion
                    _resultsListView.ItemsSource = GroupResult?.AssociationResults?
    .Where(r => !string.IsNullOrWhiteSpace(r.Title) &&
                r.Title.Trim().IndexOf(searchText, StringComparison.OrdinalIgnoreCase) >= 0);
```

Reply:
Hm, I think `?.` is enough null check. Result title need not be transformed as we're only checking for the presence of search text.